src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -5402,34 +5402,39 @@ void  MacroAssembler::decode_heap_oop_not_null(Register dst, Register src) {
 }
 
 void MacroAssembler::encode_klass_not_null(Register r, Register tmp) {
+  BLOCK_COMMENT("encode_klass_not_null {");
   assert_different_registers(r, tmp);
   if (CompressedKlassPointers::base() != nullptr) {
     if (AOTCodeCache::is_on_for_dump()) {
       movptr(tmp, ExternalAddress(CompressedKlassPointers::base_addr()));
     } else {
-      mov64(tmp, (int64_t)CompressedKlassPointers::base());
+      movptr(tmp, (intptr_t)CompressedKlassPointers::base());
     }
     subq(r, tmp);
   }
   if (CompressedKlassPointers::shift() != 0) {
     shrq(r, CompressedKlassPointers::shift());
   }
+  BLOCK_COMMENT("} encode_and_move_klass_not_null");
```

Comment:
Copy-paste nit: this is still `encode_klass_not_null`.

Reply:
Fixed